core/src/main/java/com/linecorp/armeria/server/ServiceRequestContext.java

```diff
@@ -564,19 +565,21 @@ default String pathParam(String name) {
      */
     void addAdditionalResponseHeaders(Iterable<? extends Entry<? extends CharSequence, ?>> headers);
 
-    /**
-     * Removes all headers with the specified {@code name}.
-     *
-     * @return {@code true} if at least one entry has been removed
-     */
-    boolean removeAdditionalResponseHeader(CharSequence name);
-
     /**
      * Returns the {@link HttpHeaders} which is returned along with any other trailers when a
      * {@link Service} completes an {@link HttpResponse}.
      */
     HttpHeaders additionalResponseTrailers();
 
+    /**
+     * Returns the {@link HttpHeaders} which is set using {@code (add|set)AdditionalResponseTrailer} and
+     * clears the {@link HttpHeaders} so that the {@link HttpHeaders} is not sending twice when
+     * a {@link Service} completes an {@link HttpResponse}.
+     *
+     * <p>This is used only when the trailers is sent as a {@link ResponseHeaders} in gRPC.
+     */
+    HttpHeaders getAndRemoveAdditionalResponseTrailers();
```

Comment:
I don't think we should add a gRPC only API to request context interface. Actually I was expecting a gRPC only implementation to look a little hacky hence encouraging a client option or something similarly at the client level instead :)

Reply:
That being said if we can expose this through an internal API that relies on downcast to `DefaultRequestContext` it's still ok

Reply:
Ah, Thanks. Yes, I was a little worried about it but just didn't make sure. 😉 Let me remove this.